keylime/common/algorithms.py

```diff
@@ -12,47 +12,30 @@ def is_accepted(algorithm: str, accepted: List[Any]) -> bool:
     return algorithm in accepted
 
 
-def _hashit(algorithm: str, data: bytes) -> bytes:
-    if algorithm == "sha1":
-        return hashlib.sha1(data).digest()
-    if algorithm == "sha256":
-        return hashlib.sha256(data).digest()
-    if algorithm == "sha384":
-        return hashlib.sha384(data).digest()
-    if algorithm == "sha512":
-        return hashlib.sha512(data).digest()
-    if algorithm == "sm3_256":
-        # SM3 is not guaranteed to be there
-        return hashlib.new("sm3", data).digest()
-
-    raise ValueError(f"Unsupported hash algorithm {algorithm}")
-
-
 class Hash(str, enum.Enum):
     SHA1 = "sha1"
     SHA256 = "sha256"
     SHA384 = "sha384"
     SHA512 = "sha512"
     SM3_256 = "sm3_256"
 
-    def __init__(self, *args):  # pylint: disable=unused-argument
-        super().__init__()
-        # Test hash to raise ValueError for unsupported hashes
-        _hashit(self.value, b"")
+    @classmethod
+    def from_algorithm(cls, algorithm: str) -> "Hash":
+        return cls(algorithm)
 
     @staticmethod
     def is_recognized(algorithm: str) -> bool:
         try:
             Hash(algorithm)
-            return True
         except ValueError:
             return False
+        return True
 
     def hash(self, data: bytes) -> bytes:
-        return _hashit(self.value, data)
+        return hashlib.new(self.value, data).digest()
```

Comment:
Does this work for `sm3` ? It looks like it would need "sm3" string but we seem to have the sm3_256 string. 

Also, do we still get a value error when someone creates an sm3_256 hash but the underlying crypto library doesn't support it? We don't want this to happen nor is_recognized() to pass and fail much later in the code on hashing.